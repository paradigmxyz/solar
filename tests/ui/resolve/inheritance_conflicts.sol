contract A {
    uint public x = 0;
}

contract B is A {
    uint public x = 1; //~ ERROR identifier `x` already declared
}

contract AA {
    uint public y = 2;
}

contract BB {
    uint public y = 3; //~ ERROR identifier `y` already declared
}

contract CC is AA, BB {}
