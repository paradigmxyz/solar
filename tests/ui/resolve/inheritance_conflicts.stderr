error: identifier `x` already declared
  --> ROOT/tests/ui/resolve/inheritance_conflicts.sol:LL:CC
   |
LL |     uint public x = 0;
   |                 - note: previous declaration declared here
LL | }
LL |
LL | contract B is A {
LL |     uint public x = 1;
   |                 ^
   |

error: identifier `y` already declared
  --> ROOT/tests/ui/resolve/inheritance_conflicts.sol:LL:CC
   |
LL |     uint public y = 2;
   |                 - note: previous declaration declared here
LL | }
LL |
LL | contract BB {
LL |     uint public y = 3;
   |                 ^
   |

error: aborting due to 2 previous errors

